<a href="https://colab.research.google.com/github/barcellfe/datascience_study/blob/master/Teste_COVID_D%C2%B4Vinci3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Inicialização Básica do Sandbox - Importação de Bibliotecas

**Preparando o ambiente**
inicializando bibliotecas


In [3]:
import pandas as pd
import numpy as np
import random

In [1]:
!pip install matplotlib

# 1a Atividade - Buscando resolver exercício sobre COVID 19 porém com BD da WHO

**Teste para uma Análise de Dados Exploratória (EDA) com os dados disponibilizados no site da WHO em 22 de Agosto de 2020** 

Fonte de dados: https://covid19.who.int/table  

**Questões a respnder**



1. Quantos registros existem na tabela?
2. Quantos países existem nessa base?
3. Quantos mortos, até a data do arquivo, na China?
4. Qual o total de mortes, até a data do araquivo, no mundo todo?
5. Que país teve mais casos até a data do arquivo?
6. Montando uma tabela por data para construir um gráfico de progressão comparando cinco países

*1. - Quantos registros existem na tabela?*

**Lendo o arquivo com os dados da WHO**


In [8]:
# Código abaixo utilizado para quando importamos o csv para o nosso colab, com o API / link da WHO disponibilizado isso ñ é mais necessário
# Who_Cov_Data = pd.read_csv('/content/drive/My Drive/Colab Notebooks/WHO-COVID-19-global-data-22-08-2020.csv')

Who_Cov_Data = pd.read_csv('https://covid19.who.int/WHO-COVID-19-global-data.csv')
Who_Cov_Data.head()


Date_reported  Country_code  ...  New_deaths  Cumulative_deaths
0    2020-02-24            AF  ...           0                  0
1    2020-02-25            AF  ...           0                  0
2    2020-02-26            AF  ...           0                  0
3    2020-02-27            AF  ...           0                  0
4    2020-02-28            AF  ...           0                  0

[5 rows x 8 columns]

In [9]:
Who_Cov_Data.shape

(36625, 8)

In [10]:
# Questão 1 - Resposta: São 36.625 registros em 8 colunas 
# Análise exploratória inicial - são todos os registros Not Null? 
Who_Cov_Data.count()

Date_reported         36625
 Country_code         36463
 Country              36625
 WHO_region           36625
 New_cases            36625
 Cumulative_cases     36625
 New_deaths           36625
 Cumulative_deaths    36625
dtype: int64

In [11]:
# Identificamos que a Coluna Country Code ñ possue a mesma qtd de reg das outras colunas

Who_Cov_Data.isnull().sum()

# 162 refistros nulas na coluna Country COde

Date_reported           0
 Country_code         162
 Country                0
 WHO_region             0
 New_cases              0
 Cumulative_cases       0
 New_deaths             0
 Cumulative_deaths      0
dtype: int64

**Questão 1** *Resposta* - Existem  36.625  registros nessa base. 
**Nota: Existem 162 registros com o campo Country_code igual a "Null"**

**Questão 2 - Quantos países existem nessa base?**
_______________________________________________________________________________

In [12]:
# Alterando nomes cabeçalho das colunas para que sejan str

Who_Cov_Data.columns = ['Data','Codigo Pais','Pais','Regiao','Casos Novos','Casos Acumulados','Novas Mortes','Mortes Acumuladas']
Who_Cov_Data.head()

Data Codigo Pais  ... Novas Mortes Mortes Acumuladas
0  2020-02-24          AF  ...            0                 0
1  2020-02-25          AF  ...            0                 0
2  2020-02-26          AF  ...            0                 0
3  2020-02-27          AF  ...            0                 0
4  2020-02-28          AF  ...            0                 0

[5 rows x 8 columns]

In [14]:
# COntagem de qtos países diferentes existem no DF

Conta_Pais = Who_Cov_Data['Pais'].drop_duplicates()
Conta_Pais.count()

216

In [15]:
# Identificando que países estão com Country Code = null

Who_Cov_Data[Who_Cov_Data['Codigo Pais'].isnull()]['Pais'].drop_duplicates()

23077    Namibia
Name: Pais, dtype: object

**Questão 2** *Resposta - * Existem 216 países nessa base. Namibia é o pais cujo o Country Code está como nulo - 162 ocorrências nessa base de dados

**Questão 3 - Quantos mortos, até a data do arquivo, na China?**

In [18]:
# Criando um DF apartir do BD com os registros cujo country = china

Who_Cov_Data_China = Who_Cov_Data[Who_Cov_Data.Pais=="China"]
Who_Cov_Data_China.head()

Data Codigo Pais  ... Novas Mortes Mortes Acumuladas
6917  2020-01-04          CN  ...            0                 0
6918  2020-01-05          CN  ...            0                 0
6919  2020-01-06          CN  ...            0                 0
6920  2020-01-07          CN  ...            0                 0
6921  2020-01-08          CN  ...            0                 0

[5 rows x 8 columns]

In [19]:
# Identificando quantos registros para China

Who_Cov_Data_China.shape

(232, 8)

In [25]:
# Uma forma rápida de responder é ordenando os dados do reg mais antigo ao mais recente e verificar valor da coluna Mortes Acumuladas

Who_Cov_Data_China = Who_Cov_Data_China.sort_values(by = 'Data')
Who_Cov_Data_China

Data Codigo Pais  ... Novas Mortes Mortes Acumuladas
6917  2020-01-04          CN  ...            0                 0
6918  2020-01-05          CN  ...            0                 0
6919  2020-01-06          CN  ...            0                 0
6920  2020-01-07          CN  ...            0                 0
6921  2020-01-08          CN  ...            0                 0
...          ...         ...  ...          ...               ...
7144  2020-08-18          CN  ...            0              4710
7145  2020-08-19          CN  ...            2              4712
7146  2020-08-20          CN  ...            1              4713
7147  2020-08-21          CN  ...            3              4716
7148  2020-08-22          CN  ...            0              4716

[232 rows x 8 columns]

In [27]:
Who_Cov_Data_China_Morte = Who_Cov_Data_China[Who_Cov_Data_China.Data=='2020-08-22']
Who_Cov_Data_China_Morte.head()

Data Codigo Pais  ... Novas Mortes Mortes Acumuladas
7148  2020-08-22          CN  ...            0              4716

[1 rows x 8 columns]

In [33]:
print('O No de mortes acumuladas na {} é de {}'.format(Who_Cov_Data_China_Morte['Pais'],Who_Cov_Data_China_Morte['Mortes Acumuladas'])) 

O No de mortes acumuladas na 7148    China
Name: Pais, dtype: object é de 7148    4716
Name: Mortes Acumuladas, dtype: int64


**Questão 3** *Resposta -* São 4.716 mortos, até a data do exercício, 2020-08-22, na China.
**Embora muito baixo com o que está sendo reportado no site da WHO**